CMake 4.0 removed compatibility with CMake < 3.5

### Expected behavior

Successful compilation with `CMake 4.0`

### Actual behavior

`CMake` removed with their newest release (`4.0.0`) compatibility with `CMake < 3.5`.
It can be overriden, but best it would be addressed upstream. Especially if the compatibility was/is necessary.

### Steps to reproduce

Compile the source with CMake 4.0

```
CMake Error at CMakeLists.txt:5 (cmake_minimum_required):
  Compatibility with CMake < 3.5 has been removed from CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.

  Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.
```

### Environment

**Operating system:**

Linux

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@
 #
 # SPDX-License-Identifier: GPL-2.0-or-later
 
-cmake_minimum_required (VERSION 3.0)
+cmake_minimum_required (VERSION 3.10)
 
 message ("-- Configuring the Scanner...")
 

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -71,7 +71,7 @@ if (NOT CMAKE_BUILD_TYPE MATCHES "Release")
     add_custom_target (revisiontag ALL)
 
     # creates gitversion.h using cmake script
-    add_custom_command (TARGET revisiontag COMMAND ${CMAKE_COMMAND}
+    add_custom_command (TARGET revisiontag POST_BUILD COMMAND ${CMAKE_COMMAND}
                         -DSOURCE_DIR=${CMAKE_SOURCE_DIR}
                         -P ${CMAKE_SOURCE_DIR}/cmake/GetGit.cmake)
 

